src/api/api.ts

```diff
@@ -12,41 +12,39 @@ import {
     StudentResponse,
     SubmissionResponse,
     ServerSettingsResponse,
+    InstructorResponse,
 } from './api-responses'
+import { IInstructor, Instructor } from './instructor'
+
+export interface UpdateAssignmentData {
+    new_name?: string | null,
+    directory_path?: string | null,
+    available_date?: string | null,
+    due_date?: string | null
+}
 
 export interface GetAssignmentsResponse {
     assignments: IAssignment[] | null
     currentAssignment: ICurrentAssignment | null
 }
 
-export interface GetStudentAndCourseResponse {
-    student: IStudent
+export interface GetInstructorAndStudentsAndCourseResponse {
+    instructor: IInstructor
+    students: IStudent[]
     course: ICourse
 }
 
-export async function getStudentAndCourse(): Promise<GetStudentAndCourseResponse> {
-    const { student, course } = await requestAPI<{
-        student: StudentResponse
+export async function getInstructorAndStudentsAndCourse(): Promise<GetInstructorAndStudentsAndCourseResponse> {
```

Comment:
I would just call this `getCourseMetadata`, since the information about the instructor, students, and course amounts to the course metadata right? Seems like a much more easily readable name too.

Reply:
Yeah at this point it should be lol, it started as a single endpoint at 1 point and kept getting more stuff added...